src/main/java/com/zuzka/myshowcode/entity/Item.java

```diff
@@ -0,0 +1,18 @@
+package com.zuzka.myshowcode.entity;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import javax.persistence.Embeddable;
+
+@Embeddable
```

Comment:
Nesouhlasím s tímhle návrhem. `@Embeddable` se používá, pokud někde chceš přepoužívat nějakou strukturu. Např. pokud máš tabulku Employee a Customer, oba dva mají společné např. jméno, příjmení, titul, věk, atd... Pak tedy má smysl udělat `@Embeddable` Person s těmito fieldy a dát to do Employee a Customer. I tak je použití Embeddable spekulativní, protože je smysluplnější leckdy udělat rovnou entitu Person a mít u ní např. enum pro Employee nebo Customer.

Item má referenci na Product, tato reference by tedy měla být zachována skrze joiny pro zajištění nějaké konzistence. To by znamenalo mít v Order @OneToMany relationship na OrderItem, který by měl svojí tabulku.